### PR TITLE
Added homebrew (package manager) support

### PR DESCRIPTION
**Pull request by cedricloneux**
I added a formula file that allow to install and manage the CLI from macOS most popular package manager: Homebrew. I also updated README.md with install instructions.

To allow the users to upgrade via the package manager when a new version is released, the version number and sha256sum variables should be updated. I highlighted these variables in the file.
  I might do a pull request later with a GitHub workflow to automate this task.  

Feel free to ask me any questions about this. 😃

### CODE REVIEW

**Comment by ReagentX** [comment]
How will this work on Linux or other platforms that `brew` supports?

File: README.md [hunk 1]
```diff
@@ -8,8 +8,19 @@ This crate provides both a library to interact with iMessage data as well as a b
 
 The `imessage-exporter` binary exports iMessage data to `txt` or `html` formats. It can also run diagnostics to find problems with the iMessage database.
 
+### Installation
```
Comment (ReagentX) [inline README.md:11]: Please revert this change, I want to keep all of the install stuff in one place.

**Comment by cedricloneux** [comment]
The installation procedure is the same for all plateforms

File: README.md [hunk 2]
```diff
@@ -8,8 +8,19 @@ This crate provides both a library to interact with iMessage data as well as a b
...
+
+#### From source
 Installation instructions for the binary are located [here](imessage-exporter/README.md).
 
+
+#### Using Homebrew (macOS)
```
Comment (ReagentX) [inline README.md:17]: This section belongs here: https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#installing-manually

Below cargo, above manual

File: imessage-exporter.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  binary_name = "imessage-exporter-" + (Hardware::CPU.type == :intel ? "x86_64" : "aarch64") + "-apple-darwin.tar.gz"
+  binary_url = "#{repo_url}/releases/latest/download/#{binary_name}"
+  url binary_url
+  sha256 Hardware::CPU.type == :intel ? sha256sum_intel : sha256sum_arm64
+
+  def install
+    bin.install "imessage-exporter"
+  end
+
+  test do
```
Comment (ReagentX) [inline imessage-exporter.rb:22]: Paste in the tests I wrote here: https://github.com/ReagentX/imessage-exporter/issues/26#issue-1415666265

**Comment by ReagentX** [comment]
> The installation procedure is the same for all plateforms

**Comment by ReagentX** [comment]
Sorry, hit the wrong button

**Comment by ReagentX** [comment]
> The installation procedure is the same for all plateforms

Right, but the binaries are only for `apple-darwin`, and `brew` can run on other platforms besides macOS. Will this install a non-functioning binary on a Windows/Linux machine?

**Comment by cedricloneux** [comment]
Brew is not available for windows but for now it works for macOS as you only have macOS binaries in release I used that. Do you plan to add Linux in release ? Does building from source currently works for Linux ?

File: README.md (hunk 2 above)
Comment (cedricloneux) [reply README.md:17]: You should consider putting installation instructions in the main README so everyone can see it.
It's hard to find it like this.

**Comment by cedricloneux** [comment]
I can adapt to script to dynamically detect the platform as I do for arch and from that set the filename. But for that you need to create releases for linux.

**Comment by ReagentX** [comment]
> Do you plan to add Linux in release? 

Not at the moment.

> Does building from source currently works for Linux?

Yes.

***

My worry with this is the release workflow. Currently I create a tag/release, then run build.sh to publish the crates, create the binaries, and upload them to the release page. If I use this file, I have to the make a subsequent PR to update this file. It's doable but pesky.

**Comment by cedricloneux** [comment]
It's up to you to decide your workflow. You could create a makefile or justfile with actions like make, make install.
I think continuous integrations could solve the problem. For instance, use GitHub actions.
When you make a git tag and push you use name such as "v1.0" or "1.0.0-rc1" so it's possible to see if it's pre-release or release. Github will automatically generate it. Github actions could then automatically build binaries and add them to assets and update the formula file to update the version number and sha256sum.
Maybe this actions could also build linux binaries.
This way the workflow seems easy and available for all platforms. You just have to add a tag with a correct name and push

File: README.md (hunk 2 above)
Comment (ReagentX) [reply README.md:17]: I have a local branch where I am reformatting a bunch of the readme stuff, I want to have as few changes as possible for when I merge it in. The changes I requested will make my merges easier.

**Comment by ReagentX** [comment]
Closing this because this software is now in the main `brew` index: https://formulae.brew.sh/formula/imessage-exporter

**Comment by cedricloneux** [comment]
is a json file required now ? Did homebrew stopped using ruby ?
[Your formula file](https://formulae.brew.sh/api/formula/imessage-exporter.json):  looks insanly complex compared to the version I submitted.
Why is there a different binary and hash for each macOS version ? Shouldn't it be the same?
 I even made a simpler one locally using classic ruby (tests are from you). Here it is:

```ruby
class ImessageExporter < Formula
  # change these var when a new version is released
  version "1.0.0"
  sha256 arm:   "c0a20681d6c90a5d22314d585364de8d62def9579c8014acbe4bcaa13d6d23b6",
         intel: "4d7f790d1da616e08cbd71a390362b1cf311b17a77386ee0ea9fe147944ab914"

  # formula infos
  arch arm: "aarch64", intel: "intel"
  desc "CLI to export MacOS iMessage data + run diagnostics"
  repo_url = "https://github.com/ReagentX/imessage-exporter"
  homepage repo_url

  binary_url = "#{repo_url}/releases/download/#{version}/imessage-exporter-#{arch}-apple-darwin.tar.gz"
  url binary_url
      verified: repo_url

  def install
    bin.install "imessage-exporter"
  end

  test do
    assert_equal "imessage is not a valid export type! Must be one of <txt, html>\n",
    shell_output("imessage-exporter -f imessage")
    assert_equal "Diagnostics are enabled; format is disallowed\n",
    shell_output("imessage-exporter -f txt -d")
    assert_equal "No export type selected, required by no-copy\n",
    shell_output("imessage-exporter -n")
    assert_equal "No export type selected, required by export-path\n",
    shell_output("imessage-exporter -o imessage")
    assert_equal "",
    shell_output("imessage-exporter -p imessage")
  end
end
```

Do you plan to fix the to dev workflow to make a CI/CD pipeline as I suggested multiple times?
That would reduce the homebrew formula maintenance to zero. Here you have to manually update it and I can already see it's not up to date. Last version on home-brew is 1.0.0 but it's 1.2 from repo
